src/domain/file/command.ts

```diff
@@ -0,0 +1,45 @@
+import type { IOEither } from 'fp-ts/IOEither'
+
+export type MediaType = string
+
+export type FileId = string
+
+export type StoreFileInput = {
+  id: FileId
+  name: string
+  path: string
+  size: number
+  type: MediaType
+  stream: ReadableStream
+}
+
+export type StoreFilesInput = StoreFileInput[]
+
+export const ResourceConflictError = () =>
+  ({
+    _tag: 'resource-conflict'
+  } as const)
+
+/**
+ *  Error thrown when an id of a file to be stored already exists in memory,
+ *  or if multiple files in the StoreFileInput payload have the same id
+ */
+export type ResourceConflictError = ReturnType<typeof ResourceConflictError>
+
+export const ResourceNotFoundError = (fileId: FileId) =>
+  ({
+    _tag: 'resource-not-found',
+    fileId
+  } as const)
+
+/**
+ *  Error thrown when the id of a file to be removed is not found in the list of stored files in memory.
+ */
+export type ResourceNotFoundError = ReturnType<typeof ResourceNotFoundError>
```

Comment:
As an improvement, it might be interesting to make them instances of Error or custom class that extends Error. Thus include stack trace and additional debugging information.

Reply:
@0xLodz 
We have decided to not use `Error` classes by switching to opaque types (partially) that can carry, in addition to the `tag`, an infinity of additional properties like custom messages etc.